merkle-tree/src/oneshot.rs

```diff
@@ -5,19 +5,73 @@ use simperby_common::MerkleProof;
 ///
 /// This is useful for per-block data such as transaction lists.
 pub struct OneshotMerkleTree {
-    // TODO
+    hash_list: Vec<Hash256>,
 }
 
 impl OneshotMerkleTree {
-    pub fn create(_data: Vec<Hash256>) -> Self {
-        unimplemented!()
+    pub fn create(data: Vec<Hash256>) -> Self {
+        OneshotMerkleTree { hash_list: data }
     }
 
-    pub fn create_merkle_proof(&self, _key: Hash256) -> MerkleProof {
-        unimplemented!()
+    pub fn create_merkle_proof(&self, key: Hash256) -> MerkleProof {
+        let proof: MerkleProof = MerkleProof { proof: Vec::new() };
+
+        Self::merkle_proof(&self.hash_list, key, proof)
+    }
+
+    fn merkle_proof(
```

Comment:
I think this function should return a list of `Hash256` and `MerkleProof` is constructed with the returned value in `pub fn create_merkle_proof(&self, key: Hash256) -> MerkleProof`. If you intended to implement the tail recursion, I guess just using an explicit loop is better.

Reply:
Applied